### PR TITLE
some bug fix when enable the EXEC_EnV_OLS

### PR DESCRIPTION
**Pull request by jinwei14**
- [X] Does this PR have a descriptive title that could go in our release notes?
- [ ] Does this PR add any new dependencies?
- [ ] Does this PR modify any existing APIs?
   - [ ] Is the change to the API backwards compatible?
- [X] Should this result in any changes to our documentation, either updating existing docs or adding new ones?
 
#### What does this implement/fix? Briefly explain your changes.
fixes some error when enabling EXEC_ENV_OLS

### CODE REVIEW

File: src/index.cpp [hunk 1]
```diff
@@ -517,12 +517,12 @@ void Index<T, TagT, LabelT>::load(const char *filename, uint32_t num_threads, ui
     _has_built = true;
 
     size_t tags_file_num_pts = 0, graph_num_pts = 0, data_file_num_pts = 0, label_num_pts = 0;
-
+#ifndef EXEC_ENV_OLS
```
Comment (gopalrs) [inline src/index.cpp:520]: I assume you are not looking to have filter support in the BANN code yet?

File: src/in_mem_data_store.cpp [hunk 2]
```diff
@@ -43,25 +42,25 @@ template <typename data_t> location_t InMemDataStore<data_t>::load(const std::st
...
     size_t file_dim, file_num_points;
 
     diskann::get_bin_metadata(reader, file_num_points, file_dim);
 
-    if (file_dim != _dim)
+    if (file_dim != _aligned_dim)
     {
         std::stringstream stream;
-        stream << "ERROR: Driver requests loading " << _dim << " dimension,"
+        stream << "ERROR: Driver requests loading " << _aligned_dim << " dimension,"
```
Comment (gopalrs) [inline src/in_mem_data_store.cpp:54]: This should also be "_dim"

File: include/index.h [hunk 3]
```diff
@@ -325,8 +325,10 @@ template <typename T, typename TagT = uint32_t, typename LabelT = uint32_t> clas
     // Graph related data structures
     std::vector<std::vector<uint32_t>> _final_graph;
 
+    T *_data = nullptr; // coordinates of all base points
     // Dimensions
     size_t _dim = 0;
+    size_t _aligned_dim;
```
Comment (gopalrs) [inline include/index.h:331]: We prefer not to have aligned dim in the index class and instead ask the datastore for it

File: src/in_mem_data_store.cpp [hunk 4]
```diff
@@ -43,25 +42,25 @@ template <typename data_t> location_t InMemDataStore<data_t>::load(const std::st
 }
 
 #ifdef EXEC_ENV_OLS
-template <typename data_t> location_t Index<data_t>::load_impl(AlignedFileReader &reader)
+template <typename data_t> location_t InMemDataStore<data_t>::load_impl(AlignedFileReader &reader)
```
Comment (gopalrs) [inline src/in_mem_data_store.cpp:45]: The DLVS repo has the code to read an index file using an AlignedFileReader. Please check [that implementation ](https://msasg.visualstudio.com/Bing_and_IPG/_git/ObjectStore?path=/private/Pegasus/src/DssmSearch/randnsg/src/index.cpp)  for the load_impl function that takes an AlignedFileReader as argument.

As of now, this function is not loading the index.

**Review by gopalrs — CHANGES_REQUESTED** [review]

File: include/filter_utils.h [hunk 5]
```diff
@@ -79,6 +79,7 @@ inline tsl::robin_map<std::string, std::vector<uint32_t>> generate_label_specifi
     path input_data_path, tsl::robin_map<std::string, uint32_t> labels_to_number_of_points,
     std::vector<label_set> point_ids_to_labels, label_set all_labels)
 {
+#ifndef _WINDOWS
```
[the remark this answers is not in the document]
Comment (yashpatel007) [reply include/filter_utils.h:82]: Sure. Let me take a closer look.

[the remark this answers is not in the document]
Comment (yashpatel007) [reply include/filter_utils.h:82]: I remember now, generate_label_specific_vector_files_compat() is for windows and generate_label_specific_vector_files() is for linux. Do we want to name both of them same ?

**Review by gopalrs — APPROVED** [review]

**Pull request merged**